packages/twenty-front/src/modules/ui/layout/modal/components/__stories__/Modal.stories.tsx

```diff
@@ -29,8 +64,63 @@ export const Default: Story = {
       </>
     ),
   },
-  decorators: [ComponentDecorator],
-  argTypes: {
-    children: { control: false },
+};
+
+export const CloseClosableModalOnClickOutside: Story = {
+  args: {
+    modalId: 'modal-id',
+    size: 'medium',
+    padding: 'medium',
+    isClosable: true,
+    onClose: closeMock,
+    children: (
+      <>
+        <Modal.Header>Click Outside Test</Modal.Header>
+        <Modal.Content>
+          This modal should close when clicking outside of it.
+        </Modal.Content>
+      </>
+    ),
+  },
+  play: async ({ canvasElement }) => {
+    const canvas = within(canvasElement);
+
+    await canvas.findByText('Click Outside Test');
+
+    closeMock.mockClear();
+
+    const backdrop = document.querySelector('.modal-backdrop') as HTMLElement;
+    await userEvent.click(backdrop);
+
+    expect(closeMock).toHaveBeenCalledTimes(1);
```

Comment:
I think it would be safer to use `waitFor`

```suggestion
    await waitFor(() => expect(closeMock).toHaveBeenCalledTimes(1))
```